timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/database/tinkerpop/Neo4jIndexHandler.java

```diff
@@ -46,15 +50,32 @@ public boolean hasQuickSearchIndexFor(Collection collection) {
 
   @Override
   public void addToQuickSearchIndex(Collection collection, String quickSearchValue, Vertex vertex) {
-    Index<Node> index = getFulltextIndex(collection.getCollectionName());
+    Index<Node> index = getFulltextIndex(getIndexName(collection));
 
-    index.add(graphDatabase().getNodeById((long) vertex.id()), QUICK_SEARCH, quickSearchValue);
+    index.add(vertexToNode(vertex), QUICK_SEARCH, quickSearchValue);
   }
 
-  private GraphDatabaseService graphDatabase() {
-    return tinkerpopGraphManager.getGraphDatabase();
+  private GraphTraversal<Vertex, Vertex> traversalFromIndex(Collection collection, QuickSearch quickSearch) {
+    Index<Node> index = getFulltextIndex(getIndexName(collection));
+    IndexHits<Node> hits = index.query(QUICK_SEARCH, createQuery(quickSearch));
+    List<Long> ids = StreamSupport.stream(hits.spliterator(), false).map(h -> h.getId()).collect(toList());
+
+    return ids.isEmpty() ? EmptyGraphTraversal.instance() : traversal().V(ids);
```

Comment:
will this work? V accepts a splat-array (Object...) and you're passing it a List instead of an array